modules/generative-databricks/clients/databricks.go

```diff
@@ -0,0 +1,363 @@
+		Content: prompt,
+	}}
+
+	input = generateInput{
+		Messages:    messages,
+		Stream:      false,
+		Logprobs:    params.Logprobs,
+		TopLogprobs: params.TopLogprobs,
+		MaxTokens:   params.MaxTokens,
+		N:           params.N,
+		Stop:        params.Stop,
+		Temperature: params.Temperature,
+		TopP:        params.TopP,
+	}
+
+	return input, nil
+}
+
+func (v *databricks) getError(statusCode int, resBodyError *databricksApiError) error {
+	endpoint := "Databricks Foundation Model API"
+	if resBodyError != nil {
+		return fmt.Errorf("connection to: %s failed with status: %d error: %v", endpoint, statusCode, resBodyError.Message)
+	}
+	return fmt.Errorf("connection to: %s failed with status: %d", endpoint, statusCode)
+}
+
+func (v *databricks) getApiKeyHeaderAndValue(apiKey string) (string, string) {
+	return "Authorization", fmt.Sprintf("Bearer %s", apiKey)
+}
+
+func (v *databricks) generatePromptForTask(textProperties []map[string]string, task string) (string, error) {
+	marshal, err := json.Marshal(textProperties)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf(`'%v:
+%v`, task, string(marshal)), nil
+}
+
+func (v *databricks) generateForPrompt(textProperties map[string]string, prompt string) (string, error) {
+	all := compile.FindAll([]byte(prompt), -1)
+	for _, match := range all {
+		originalProperty := string(match)
+		replacedProperty := compile.FindStringSubmatch(originalProperty)[1]
+		replacedProperty = strings.TrimSpace(replacedProperty)
+		value := textProperties[replacedProperty]
+		if value == "" {
+			return "", errors.Errorf("Following property has empty value: '%v'. Make sure you spell the property name correctly, verify that the property exists and has a value", replacedProperty)
+		}
+		prompt = strings.ReplaceAll(prompt, originalProperty, value)
+	}
+	return prompt, nil
+}
+
+func (v *databricks) getApiKey(ctx context.Context) (string, error) {
+	var apiKey, envVarValue, envVar string
+
+	apiKey = "X-Databricks-Token"
+	envVar = "DATABRICKS_TOKEN"
+	envVarValue = v.databricksToken
+
+	return v.getApiKeyFromContext(ctx, apiKey, envVarValue, envVar)
+}
+
+func (v *databricks) getApiKeyFromContext(ctx context.Context, apiKey, envVarValue, envVar string) (string, error) {
+	if apiKeyValue := v.getValueFromContext(ctx, apiKey); apiKeyValue != "" {
+		return apiKeyValue, nil
+	}
+	if envVarValue != "" {
+		return envVarValue, nil
+	}
+	return "", fmt.Errorf("no api key found neither in request header: %s nor in environment variable under %s", apiKey, envVar)
+}
+
+func (v *databricks) getValueFromContext(ctx context.Context, key string) string {
+	if value := ctx.Value(key); value != nil {
+		if keyHeader, ok := value.([]string); ok && len(keyHeader) > 0 && len(keyHeader[0]) > 0 {
+			return keyHeader[0]
+		}
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
+
+	return ""
```

Comment:
instead of all of this code you can use `modulecomponents.GetValueFromContext` method

Reply:
nice! thanks for sharing!